src/Extension/Footnote/Parser/AnonymousFootnoteRefParser.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+/*
+ * This file is part of the league/commonmark package.
+ *
+ * (c) Colin O'Dell <colinodell@gmail.com>
+ * (c) Rezo Zero / Ambroise Maupate
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace League\CommonMark\Extension\Footnote\Parser;
+
+use League\CommonMark\Extension\Footnote\Node\FootnoteRef;
+use League\CommonMark\Inline\Parser\InlineParserInterface;
+use League\CommonMark\InlineParserContext;
+use League\CommonMark\Reference\Reference;
+
+final class AnonymousFootnoteRefParser implements InlineParserInterface
+{
+    public function getCharacters(): array
+    {
+        return ['^'];
+    }
+
+    public function parse(InlineParserContext $inlineContext): bool
+    {
+        $container = $inlineContext->getContainer();
+        $cursor = $inlineContext->getCursor();
+        $nextChar = $cursor->peek();
+        if ($nextChar !== '[') {
+            return false;
+        }
+        $state = $cursor->saveState();
+
+        $m = $cursor->match('/\^\[[^\n^\]]+\]/');
+        if ($m !== null) {
+            if (\preg_match('#\^\[([^\]]+)\]#', $m, $matches) > 0) {
+                $reference = $this->createReference($matches[1]);
+                $container->appendChild(new FootnoteRef($reference, $matches[1]));
+
+                return true;
+            }
+        }
+
+        $cursor->restoreState($state);
+
+        return false;
+    }
+
+    private function createReference(string $label): Reference
+    {
+        $refLabel = Reference::normalizeReference($label);
+        $refLabel = \mb_strtolower(\str_replace(' ', '-', $refLabel));
```

Comment:
FWIW, this is why I was originally saying that creating a more generic `StubGenerator` (that isn't tied to headings) might be useful; for cases like this.

Reply:
You should cleanup reference not to get weird HTML class and id : 

https://github.com/rezozero/commonmark-ext-footnotes/blob/72afb206b9b804c7d788815e56af2caf37d6bd1c/src/Parser/AnonymousFootnoteRefParser.php#L47

Reply:
That's why I was suggesting using the `SlugGenerator`, but moving it out of the `HeadingPermalink` namespace and into a more generic one since it could be used by multiple things: https://github.com/thephpleague/commonmark/blob/f40f6194960c60893537b86bf64f2f3369b72f01/src/Extension/HeadingPermalink/SlugGenerator/DefaultSlugGenerator.php#L33-L45

Also, it probably wouldn't hurt to have a similar `slug_generator` option for this extension as well.

Reply:
In #481 we changed the interface to accept a `Node` instead of a `string` in order to provide extra context about the heading node:

https://github.com/thephpleague/commonmark/blob/f40f6194960c60893537b86bf64f2f3369b72f01/src/Extension/HeadingPermalink/SlugGenerator/SlugGeneratorInterface.php#L16-L22

This suggestion sounds like we want to lean back towards using the text as the primary source of information for the slug.

What if we moved this out of the `HeadingPermalink` namespace as suggested and changed the method signature to look more like this:

```php
    /**
     * @param string $text    The text to slugify
     * @param mixed  $context Additional context about the text being slugified (optional)
     */
    public function generateSlug(string $text, $context = null): string;
```

In the case of heading permalinks, `$context` would receive the `Heading` node.  In this particular case we only have a string, so `$context` would be `null`.  And in other potential future cases, we might have `$context` be some other object, or perhaps an array of relevant items.

What would you think about something like that @markcarver?  I'm open to other ideas as well.

Reply:
(Personally I don't like having the `mixed` type here, but I feel like we either do that or have separate interfaces for slugifying strings vs nodes vs whatever else we might have in the future)

Reply:
@colinodell IIRC, that was my original suggestion, just passing an extra parameter for context. In that case, it was the Heading... I'm assuming we wouldn't pass anything here? Unless we wanted to pass the inline parser(s) as well, just so one knows what requested the slug generation.

What if it had two methods (for ease of use):

```php
    public function generateSlug(string $text, $context = null): string;
    public function generateSlugFromNode(Node $node): string;
```

The latter invoking the former, and passing the `Node` as `$context`?

Reply:
> IIRC, that was my original suggestion, just passing an extra parameter for context.

Gotcha, I apologize for missing/misunderstanding that :upside_down_face: 

> In that case, it was the Heading... I'm assuming we wouldn't pass anything here?

Exactly.

> Unless we wanted to pass the inline parser(s) as well, just so one knows what requested the slug generation.

I think we'd be okay omitting that if we implement the `slug_generator` option.

Let me merge this as-is and create a separate MR for these slug-related changes :wink: